sql/mysql/sqlspec.go

```diff
@@ -256,9 +256,28 @@ func columnSpec(c *schema.Column, t *schema.Table) (*sqlspec.Column, error) {
 	if sqlx.Has(c.Attrs, &AutoIncrement{}) {
 		col.Extra.Attrs = append(col.Extra.Attrs, specutil.BoolAttr("auto_increment", true))
 	}
+	if x := (schema.GeneratedExpr{}); sqlx.Has(c.Attrs, &x) {
+		appendGenExpr(col, x)
+	}
 	return col, nil
 }
 
+// appendGenExpr appends the schema.GeneratedExpr as an attribute or a resource.
+func appendGenExpr(spec *sqlspec.Column, x schema.GeneratedExpr) {
+	t := strings.ToUpper(x.Type)
+	if t == "" || t == genVirtual {
+		spec.Extra.Attrs = append(spec.Extra.Attrs, specutil.StrAttr("as", x.Expr))
+		return
+	}
+	spec.Extra.Children = append(spec.Extra.Children, &schemaspec.Resource{
+		Type: "as",
+		Attrs: []*schemaspec.Attr{
+			specutil.VarAttr("type", t),
+			specutil.StrAttr("expr", x.Expr),
+		},
+	})
+}
+
```

Comment:
Shouldn't the internal representation be the same, regardless of what was defined in HCL? And I think being explicit is cleaner, so if we marshal it wen can always do the `Resource` way.

Reply:
Talked to it @rotemtam. I'll move it to the `specutil` after I'll need it in PostgreSQL.

Reply:
>  And I think being explicit is cleaner, so if we marshal it wen can always do the Resource way.

Agree.